packages/app/src/cli/models/extensions/specifications/function.ts

```diff
@@ -26,6 +26,7 @@ export const FunctionExtensionSchema = BaseSchema.extend({
           details: zod.string(),
         })
         .optional(),
+      handle: zod.string().optional(),
```

Comment:
zod?

Reply:
Looks like `zod` is used in CLI to declare and validate schemas. Not sure 100% on the specifics, but to my knowledge this will both declare the schema and validate when it's parsing the `.toml` file.  One thing that doesn't make sense to me is why we have `optional` on all the fields, but just followed the pattern in CLI for this here.

Reply:
TIL [zod](https://zod.dev/).

Maybe the optionality is related to the CLI publishing a blank on first creation? If so, it might go away after the CLI team changes that behaviour

Reply:
Either way, we want the `ui.handle` to be optional because not all APIs will require it or partners may not want to use that.